packages/react-native/React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -1189,86 +1189,88 @@ - (void)dispatchBlock:(dispatch_block_t)block queue:(dispatch_queue_t)queue
 
 - (void)invalidate
 {
-  if (_didInvalidate) {
-    return;
-  }
+  RCTUnsafeExecuteOnMainQueueSync(^{
+    if (_didInvalidate) {
+      return;
+    }
 
-  RCTAssertMainQueue();
-  RCTLogInfo(@"Invalidating %@ (parent: %@, executor: %@)", self, _parentBridge, [self executorClass]);
+    RCTAssertMainQueue();
```

Comment:
One thing that confuses me is that, in your stacktrace, the crash is happening in Thread 26... but this assert should force the app to be on the main thread, which is not the Thread 26... how's this possible?

Reply:
@cipolleschi Good question, that's because of actually the `RCTAssertMainQueue` only takes effects in dev build, when in the release build, it does nothing.

Reply:
That's why I added the `RCTUnsafeExecuteOnMainQueueSync` wrapper to ensure the code to run on the main thread.

Reply:
<img width="1068" alt="image" src="https://github.com/user-attachments/assets/e0df4205-59d8-499e-be31-4a2256df4cea" />
As you can see, when the `NS_BLOCK_ASSERTIONS` is defined, the `RCTAssert` macro is actually empty, and generally the `NS_BLOCK_ASSERTIONS` is defined in release build.

Reply:
That's a good explanation, but then we should see crashes in development happening because of the assertion. And IIUC, the app does not crash in development, right?

Reply:
By looking at the crash log, the JS thread is triggering the invalidation. I think that this is the root of the problem: after the JS thread detect the invalidation, we should jump on the UI thread to invalidate everything...

Reply:
I didn't encounter this crash in development, but I am not sure if it happens in it.